roles/add-networks/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: get node and master addresses
+  shell: "openstack coe cluster show {{ cluster_name }} -c master_addresses -c node_addresses -f json"
+  register: result_addresses
+
+- name: get list of servers
+  shell: "openstack server list -c ID -c Networks -f json"
+  register: result_servers
+
+- name: convert results to json
+  set_fact:
+    addresses: "{{ result_addresses.stdout | from_json }}"
+    servers: "{{ result_servers.stdout | from_json }}"
+
+- name: add masters
+  add_host:
+    hostname: "{{ item[1].ID }}"
+    ansible_host: "{{ item[0] }}"
+    ansible_user: "{{ cluster_user }}"
+    networks: "{{ item[1].Networks.split('; ') | join('\n') | regex_replace('=', ': ') | from_yaml }}"
+    groups:
+    - masters
+    - cluster
+  when: item[1].Networks is search(item[0])
+  loop: "{{ query('nested', addresses.master_addresses, servers) }}"
```

Comment:
TIL: this is how you loop in ansible 2.5. FWIW, this won't work in ansible <= 2.4.

Reply:
I've added ansible>=2.5 to to requirements.txt